### PR TITLE
v0.4.23

### PR DESCRIPTION
**Pull request by Izzette**
## Enter and backspace keys to add and remove black/white list items
- [x] Added lists UI feature:
  * When focused on non-empty input field, release enter to add new field and change focus to it
  * When focused on empty input field, release backspace to remove current field.  Does not remove an input field if it is the only one remaining in its list.  If available, changes focus to previous field, else change focus to next field.

- [x] Changed list remove button click-handler to named function (removeInputHandler), in order to be used with added feature.

- [x] [SOLVED]: There is one issue that I'm not quite sure how to resolve eloquently. The back button removes the field with the keystroke that removes the last remaining character. Of course, it is then as easy as pressing enter to create a new field and focus on it. However, it is not what I desired from the feature when I conceived it.

### CODE REVIEW

**Comment by Izzette** [comment]
There is one issue that I'm not quite sure how to resolve.  The back button removes the field with the keystroke that removes the last remaining character.  Of course, it is then as easy as pressing enter to create a new field and focus on it.  However, it is not what I desired from the feature when I conceived it.  I am not really sure how to solve that problem eloquently.

**Comment by Izzette** [comment]
###Previous issue solved with 0c6caa7  
  * Input fields only removed by backspace if they were empty before the most recent backspace-keyup
  * In order to implement the previous change, a parameter "wasEmpty" was added to lists input div

**Comment by Izzette** [comment]
### Previous issue solved with 9f50f803cd111992f45a7763663ad216ba4f3a17
  * Input fields only removed by backspace if they were empty before the most recent backspace-keyup
  * In order to implement the previous change, a parameter "wasEmpty" was added to lists input div

File: src/data/options.js [hunk 1]
```diff
@@ -135,8 +134,52 @@ function removeInput(optionWhich) {
...
+	}
+	if (removeThis.id.indexOf('_div') >= 0) {
+		removeInput(removeThis.id);
+	}
+	e.preventDefault();
+	e.stopPropagation();
+}
+
+function inputKeyUpHandler(e) {
+	var wasEmptyObject, nextFocusId, keycode = e.which, target = e.target, inputLength = target.value.length, divObject = target.parentNode, wasEmptyObj = divObject.getElementsByTagName('PARAM')[0], wasEmpty = 'true' === wasEmptyObj.value ? true : false;
```
Comment (bjornstar) [inline src/data/options.js]: This line is absurdly long and hiding some important variable assignments.

**Comment by bjornstar** [comment]
When I press enter, it should insert an empty field below where I was typing.

It should insert an empty field every time I press enter.

**Comment by Izzette** [comment]
It does insert and empty field enter is press, unless the focused field is empty.  I figured you wouldn't need more than one empty field at a time.

It doesn't insert it below, it just inserts it at the end.  I can change you want, but inserting at the end allows me to reuse existing code.

File: src/data/options.js (hunk 1 above)
Comment (Izzette) [inline src/data/options.js]: You're right, it is absurdly long.

**Comment by bjornstar** [comment]
Feel free to re-open with conflicts resolved.

**Pull request closed without merging**